packages/modules/link-modules/integration-tests/__tests__/migrations.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { MigrationsExecutionPlanner } from "../../src"
+import { MedusaModule, ModuleJoinerConfig } from "@medusajs/modules-sdk"
+import {
+  Car,
+  carJoinerConfig,
+  CarModule,
+  User,
+  userJoinerConfig,
+  UserModule,
+} from "../__fixtures__/migrations"
+import { defineLink, isObject, Modules } from "@medusajs/utils"
+import { moduleIntegrationTestRunner } from "medusa-test-utils"
+import { ILinkModule } from "@medusajs/types"
+
+jest.setTimeout(30000)
+
+MedusaModule.setJoinerConfig(userJoinerConfig.serviceName, userJoinerConfig)
+MedusaModule.setJoinerConfig(carJoinerConfig.serviceName, carJoinerConfig)
+
+moduleIntegrationTestRunner<ILinkModule>({
+  moduleName: Modules.LINK,
+  moduleModels: [User, Car],
+  testSuite: ({ dbConfig }) => {
+    describe("MigrationsExecutionPlanner", () => {
+      test("should generate an execution plan", async () => {
+        defineLink(UserModule.linkable.user, CarModule.linkable.car)
+
+        MedusaModule.getCustomLinks().forEach((linkDefinition: any) => {
+          MedusaModule.setCustomLink(
+            linkDefinition(MedusaModule.getAllJoinerConfigs())
+          )
+        })
+
+        /**
+         * Expect a create plan
+         */
+
+        let joinerConfigs = MedusaModule.getCustomLinks().filter(
+          (link): link is ModuleJoinerConfig => isObject(link)
+        )
+
+        let planner = new MigrationsExecutionPlanner(joinerConfigs, {
+          database: dbConfig,
+        })
+
+        let actionPlan = await planner.createPlan()
+        await planner.executePlan(actionPlan)
+
+        expect(actionPlan).toHaveLength(1)
+        expect(actionPlan[0]).toEqual({
+          action: "create",
+          linkDescriptor: {
+            fromModule: "User",
+            toModule: "Car",
+            fromModel: "user",
+            toModel: "car",
+          },
+          tableName: "User_user_Car_car",
+          sql: 'create table "User_user_Car_car" ("user_id" varchar(255) not null, "car_id" varchar(255) not null, "id" varchar(255) not null, "created_at" timestamptz(0) not null default CURRENT_TIMESTAMP, "updated_at" timestamptz(0) not null default CURRENT_TIMESTAMP, "deleted_at" timestamptz(0) null, constraint "User_user_Car_car_pkey" primary key ("user_id", "car_id"));\ncreate index "IDX_car_id_-8c9667b4" on "User_user_Car_car" ("car_id");\ncreate index "IDX_id_-8c9667b4" on "User_user_Car_car" ("id");\ncreate index "IDX_user_id_-8c9667b4" on "User_user_Car_car" ("user_id");\ncreate index "IDX_deleted_at_-8c9667b4" on "User_user_Car_car" ("deleted_at");\n\n',
```

Comment:
we might need to review the index name generation @carlos-r-l-rodrigues 😂

Reply:
Yeah, it's using the `simpleHash` of the table name.
we have to be aware of the limit of 63 chars to name stuff in Postgres. Maybe place some todos to manage that nicely later

Reply:
if people use very long names for their modules + entities this will cause issues at the moment,

Reply:
yes for sure, nothing to do now though